In [1]:
#Used libraries
import numpy as np # for linear algebra (multidimensional arrays)
import pandas as pd # data processing, CSV file I/O (example pd.read_csv)


# Exploratory Analysis

In [2]:
traindf = pd.read_json('train.json', orient='columns')
testdf = pd.read_json('test.json', orient='columns')

Exploring Training-set

In [3]:
#Displaying 20 first rows to discover the general data structure with some values
traindf.head(20)

cuisine     id                                        ingredients
0         greek  10259  [romaine lettuce, black olives, grape tomatoes...
1   southern_us  25693  [plain flour, ground pepper, salt, tomatoes, g...
2      filipino  20130  [eggs, pepper, salt, mayonaise, cooking oil, g...
3        indian  22213                [water, vegetable oil, wheat, salt]
4        indian  13162  [black pepper, shallots, cornflour, cayenne pe...
5      jamaican   6602  [plain flour, sugar, butter, eggs, fresh ginge...
6       spanish  42779  [olive oil, salt, medium shrimp, pepper, garli...
7       italian   3735  [sugar, pistachio nuts, white almond bark, flo...
8       mexican  16903  [olive oil, purple onion, fresh pineapple, por...
9       italian  12734  [chopped tomatoes, fresh basil, garlic, extra-...
10      italian   5875  [pimentos, sweet pepper, dried oregano, olive ...
11      chinese  45887  [low sodium soy sauce, fresh ginger, dry musta...
12      italian   2698  [Italian parsley leaves, walnuts, hot red pepp...
13      mexican  41995  [ground cinnamon, fresh cilantro, chili powder...
14      italian  31908  [fresh parmesan cheese, butter, all-purpose fl...
15       indian  24717  [tumeric, vegetable stock, tomatoes, garam mas...
16      british  34466  [greek yogurt, lemon curd, confectioners sugar...
17      italian   1420  [italian seasoning, broiler-fryer chicken, may...
18         thai   2941   [sugar, hot chili, asian fish sauce, lime juice]
19   vietnamese   8152  [soy sauce, vegetable oil, red bell pepper, ch...

In [4]:
#Information about the training-set (each attribute, its type and the number of its elements)
traindf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39774 entries, 0 to 39773
Data columns (total 3 columns):
cuisine        39774 non-null object
id             39774 non-null int64
ingredients    39774 non-null object
dtypes: int64(1), object(2)
memory usage: 932.3+ KB


we have 39774 elements in total without null ones, ides are of int64 type, cuisine and ingredients are of object type

In [14]:
print("the number of total recipes in the training set is : ", len(traindf))
print("the shape of the taining set is : ", traindf.shape, " :\n ",traindf.shape[0], "objects with ", traindf.shape[1]," attributes each")

the number of total recipes in the training set is :  39774
the shape of the taining set is :  (39774, 3)  :
  39774 objects with  3  attributes each


Test-set exploration and quick overview

In [16]:
testdf.head(20)

id                                        ingredients
0   18009  [baking powder, eggs, all-purpose flour, raisi...
1   28583  [sugar, egg yolks, corn starch, cream of tarta...
2   41580  [sausage links, fennel bulb, fronds, olive oil...
3   29752  [meat cuts, file powder, smoked sausage, okra,...
4   35687  [ground black pepper, salt, sausage casings, l...
5   38527  [baking powder, all-purpose flour, peach slice...
6   19666             [grape juice, orange, white zinfandel]
7   41217  [ground ginger, white pepper, green onions, or...
8   28753  [diced onions, taco seasoning mix, all-purpose...
9   22659  [eggs, cherries, dates, dark muscovado sugar, ...
10  21749  [pasta, olive oil, crushed red pepper, cherry ...
11  44967  [water, butter, ground sumac, ground lamb, gro...
12  42969  [curry powder, ground cumin, chicken wings, ch...
13  44883  [pasta, marinara sauce, dried basil, chicken f...
14  20827  [salt, custard powder, white sugar, eggs, marg...
15  23196  [vegetable oil cooking spray, egg whites, apri...
16  35387  [vanilla ice cream, banana liqueur, bananas, p...
17  33780  [molasses, hot sauce, baked beans, Grey Poupon...
18  19001  [chopped green chilies, sour cream, cheddar ch...
19  16526  [cold water, chicken drumsticks, chicken thigh...

The target variable is cuisine which is not present in the test-set

In [18]:
#Information about the test-set (each attribute, its type and the number of its elements)
testdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9944 entries, 0 to 9943
Data columns (total 2 columns):
id             9944 non-null int64
ingredients    9944 non-null object
dtypes: int64(1), object(1)
memory usage: 155.5+ KB


We have 9944 elements in total without null ones, ides are of int64 type, ingredients are of object type

In [24]:
print("the number of total recipes in the test-set is : ", len(testdf))
print("the shape of the test set is : ", testdf.shape, " :\n ",testdf.shape[0], "objects with ", testdf.shape[1]," attributes each")
print("the proportion of the test-set is : ", 100*len(testdf)/(len(traindf) + len(testdf)), "% of total data")

the number of total recipes in the test-set is :  9944
the shape of the test set is :  (9944, 2)  :
  9944 objects with  2  attributes each
the proportion of the test-set is :  20.000804537592018 % of total data


In [6]:
testdf.head(20)

id                                        ingredients
0   18009  [baking powder, eggs, all-purpose flour, raisi...
1   28583  [sugar, egg yolks, corn starch, cream of tarta...
2   41580  [sausage links, fennel bulb, fronds, olive oil...
3   29752  [meat cuts, file powder, smoked sausage, okra,...
4   35687  [ground black pepper, salt, sausage casings, l...
5   38527  [baking powder, all-purpose flour, peach slice...
6   19666             [grape juice, orange, white zinfandel]
7   41217  [ground ginger, white pepper, green onions, or...
8   28753  [diced onions, taco seasoning mix, all-purpose...
9   22659  [eggs, cherries, dates, dark muscovado sugar, ...
10  21749  [pasta, olive oil, crushed red pepper, cherry ...
11  44967  [water, butter, ground sumac, ground lamb, gro...
12  42969  [curry powder, ground cumin, chicken wings, ch...
13  44883  [pasta, marinara sauce, dried basil, chicken f...
14  20827  [salt, custard powder, white sugar, eggs, marg...
15  23196  [vegetable oil cooking spray, egg whites, apri...
16  35387  [vanilla ice cream, banana liqueur, bananas, p...
17  33780  [molasses, hot sauce, baked beans, Grey Poupon...
18  19001  [chopped green chilies, sour cream, cheddar ch...
19  16526  [cold water, chicken drumsticks, chicken thigh...

In [7]:
testdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9944 entries, 0 to 9943
Data columns (total 2 columns):
id             9944 non-null int64
ingredients    9944 non-null object
dtypes: int64(1), object(1)
memory usage: 155.5+ KB
